src/index.js

```diff
@@ -118,28 +164,63 @@ class ProfilerPlugin {
           resolve();
         });
 
+        const totalWantedFiles = [
+          this.profilerEnabled,
+          this.heapEnabled
+        ].filter(Boolean).length;
+
+        let filesSeen = 0;
+        archive.on('entry', () => {
+          filesSeen++;
+          if (filesSeen >= totalWantedFiles) {
```

Comment:
Why would we get more files than the number of files we want?

Reply:
Not sure. You'd rather use `===` here?

Reply:
Probably -- otherwise why is there a scenario where "we have an entry, and now there are more entries than there are wanted files"? that's what `>=` makes me think